content/news/2024-02-03-bevy-0.13/index.md

```diff
@@ -0,0 +1,1311 @@
+        ),
+    ),
+);
+```
+
+## Type-Safe Labels for the `RenderGraph`
+
+<div class="release-feature-authors">authors: @DasLixou</div>
+
+Bevy uses Rust's type system extensively when defining labels, letting developers lean on tooling to catch typos and ease refactors.
+But this didn't apply to Bevy's render graph. In the render graph, hard-coded—and potentially overlapping—strings were used to define nodes and sub-graphs.
+
+```rust
+// Before 0.13
+impl MyRenderNode {
+    pub const NAME: &'static str = "my_render_node"
+}
+```
+
+In Bevy 0.13, we're using a more robust way to name render nodes and render graphs with the help of Rust's type system and Bevy's label system.
+
+```rust
+// After 0.13
+#[derive(Debug, Hash, PartialEq, Eq, Clone, RenderLabel)]
+pub struct MyRenderLabel;
+```
+
+With those, the long paths for const-values also get shortened and cleaner which results into this
+
+```rust
+// Before 0.13
+render_app
+    .add_render_graph_node::<ViewNodeRunner<MyRenderNode>>(
+        core_3d::graph::NAME,
+        MyRenderNode::NAME,
+    )
+    .add_render_graph_edges(
+        core_3d::graph::NAME,
+        &[
+            core_3d::graph::node::TONEMAPPING,
+            MyRenderNode::NAME,
+            core_3d::graph::node::END_MAIN_PASS_POST_PROCESSING,
+        ],
+    );
+
+// After 0.13
+use bevy::core_pipeline::core_3d::graph::{Labels3d, SubGraph3d};
+
+render_app
+    .add_render_graph_node::<ViewNodeRunner<MyRenderNode>>(
+        SubGraph3d,
+        MyRenderLabel,
+    )
+    .add_render_graph_edges(
+        SubGraph3d,
+        (
+            Labels3d::Tonemapping,
+            MyRenderLabel,
+            Labels3d::EndMainPassPostProcessing,
+        ),
+    );
+```
+
+When you need dynamic labels for render nodes, those can still be achieved via e.g. tuple structs:
+
+```rust
+#[derive(Debug, Hash, PartialEq, Eq, Clone, RenderLabel)]
+pub struct MyDynamicLabel(&'static str);
+```
+
+This is particularly nice because we don't have to store strings in here: we can use integers, custom enums or any other hashable type.
+
+## Camera-Driven UI
+
+<div class="release-feature-authors">authors: @bardt, @oceantume</div>
+
+Historically, Bevy's UI elements have been scaled and positioned in the context of the primary window, regardless of the camera settings. This approach made some UI experiences like split-screen multiplayer difficult to implement, and others such as having UI in multiple windows impossible.
+
+We are now introducing a more flexible way of rendering the user interface: camera-driven UI. Each camera can now have its own UI root, rendering according to its viewport, scale factor, and a target which can be a secondary window or even a texture.
+
+This change unlocks a variety of new UI experiences, including split-screen multiplayer, UI in multiple windows, displaying non-interactive UI in a 3D world, and more.
+
+![Split-screen with independent UI roots](split-screen.png)
```

Comment:
These split screen UIs look almost exactly the same. Should the screenshot not express more clearly you can now have a different UI per camera?